allennlp/training/learning_rate_schedulers.py

```diff
@@ -134,6 +134,114 @@ def get_lr(self):
         return [scale for _ in range(len(self.base_lrs))]
 
 
+class SlantedTriangular(torch.optim.lr_scheduler._LRScheduler): # pylint: disable=protected-access
+    """
+    Implements the Slanted Triangular Learning Rate schedule with optional gradual
+    unfreezing. The schedule corresponds to first linearly increasing the learning
+    rate and annealing the learning based on a fixed ratio.
+    If we gradually unfreeze, then in the first epoch of training, only the top
```

Comment:
This will render nicer in the docs if you have a blank line between paragraphs.  You clearly intended that, but it won't show up right without the blank line.